resources/js/index.js

```diff
@@ -93,14 +73,84 @@ getInitialRecipes().then(initialRecipes => {
   })
 generateIngredients();
 
+function removeFromShoppingCart(element){
+    //Get the Text of the <p> Element of the List where the button was clicked.
+    let text = element.previousElementSibling.innerText;
+    var cart = JSON.parse(localStorage.getItem("cart"));
+    let index = cart.indexOf(text);
+    if (index > -1) cart.splice(index, 1); 
+    element.parentElement.parentElement.remove();
+    localStorage.setItem("cart", JSON.stringify(cart));
+}
+/**
+ * Generates the shopping cart html in following format:
+ * <li class="mb-2">
+        <div class="d-flex flex-row flex-row-table align-content-start ">
+            <p class="mb-1 align-self-center"><<INGREDIENT>></p>
+            <button type="button" onclick="removeFromShopnpingCart(this)" class="btn btn-outline-success ms-auto p-2" data-toggle="tooltip" data-placement="top" title="Add to Shopping List"></button>
+        </div>
+   </li>
+ */
+
+function generateShoppingCart(){
+    var shopping = document.getElementById("shopping-list");
+    var shopping_mobile = document.getElementById("shopping-list-mobile");
+    var ul = document.createElement("ul");
+    var cart = JSON.parse(localStorage.getItem("cart"));
+    if(cart == null){
```

Comment:
Add a check for empty array, not just null
```suggestion
    if (cart === null || cart.legnth === 0) {
```

Reply:
Alternatively, remove the cart from localstorage when the last item is removed